velox/exec/GroupId.cpp

```diff
@@ -105,7 +107,7 @@ RowVectorPtr GroupId::getOutput() {
 
   // Add groupId column.
   outputColumns[outputType_->size() - 1] =
-      BaseVector::createConstant((int64_t)groupingSetIndex_, numInput, pool());
+      BaseVector::createConstant((int64_t)gid, numInput, pool());
```

Comment:
I think this will be break Prestissimo. Would you check if com.facebook.presto.nativeworker.TestHiveAggregationQueries#testGroupingSets still passes?

Reply:
Yes. It may break the Prestissimo logic. Can we add a bool to check whether recalculate the gid? And then we Gluten can use existing GroupIdNode to support the Expand operator in Spark. Do you have other suggestions?

Reply:
Yes. It will break Prestissimo. 
Currently the spark will calculate the grouping() function based on the grouping_id. So we need this change in order to keep the same result with spark. How about add a custom GroupId for spark and then we can change the group id calculation ? Do you have other suggestions?

Reply:
@JkSelf I'm not sure I understand your comments. Would you clarify?